opal/corelib/array.rb

```diff
@@ -142,6 +142,8 @@ def self.try_convert(obj)
   end
 
   def &(other)
+    return [] if `self.length === 0`
+
     other = `convertToArray(other)`
```

Comment:
Thoughts on also adding `if (other.length === 0) return []`?

nit, optional: Also probably the two initial lines can be moved inside the `%x{…}` block.